### PR TITLE
egressIP: node retrieval failure is not respected, causes panic

### PR DESCRIPTION
**Pull request by tssurya**
Fixes panic seen during `checkEgressNodesReachability`

```
E0606 11:57:34.063567       1 egressip.go:2096] Node:
huirwang-0606c-m5w24-worker-2-dkjmc reachability changed, but could not
retrieve node from cache, err: node
"huirwang-0606c-m5w24-worker-2-dkjmc" not found
W0606 11:57:34.063658       1 egressip.go:1657] Unable to remove GARP
configuration on external logical switch port for egress node:
huirwang-0606c-m5w24-worker-2-dkjmc, err: object not found
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x176626d]

goroutine 487 [running]:
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn.(*Controller).checkEgressNodesReachability(0xc000171c00)
	/home/surya/go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/ovn/egressip.go:2099
+0x3ed
created by
github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn.(*Controller).initClusterEgressPolicies
	/home/surya/go/src/github.com/openshift/ovn-kubernetes/go-controller/pkg/ovn/egressip.go:1772
+0xdb
```

Signed-off-by: Surya Seetharaman <suryaseetharaman.9@gmail.com>

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/ovn/egressip.go [hunk 1]
```diff
@@ -1587,17 +1587,17 @@ func (oc *Controller) setNodeEgressReachable(nodeName string, isReachable bool)
 	}
 }
 
-func (oc *Controller) addEgressNode(egressNode *kapi.Node) error {
+func (oc *Controller) addEgressNode(egressNode string) error {
```
Comment (flavio-fernandes) [inline go-controller/pkg/ovn/egressip.go:1590]: NIT: wdyt we rename this to egressNodeName or nodeName?

File: go-controller/pkg/ovn/egressip.go [hunk 2]
```diff
@@ -1639,20 +1639,20 @@ func (oc *Controller) addEgressNode(egressNode *kapi.Node) error {
 	return nil
 }
 
-func (oc *Controller) deleteEgressNode(egressNode *kapi.Node) error {
+func (oc *Controller) deleteEgressNode(egressNode string) error {
```
Comment (flavio-fernandes) [inline go-controller/pkg/ovn/egressip.go:1642]: ditto

**Review by flavio-fernandes — COMMENTED** [review]
short of a little NIT, which you may not really have to follow on, it lgtm!
will be good to add a test to cover cases when node is nil, but that may be indeed tricky now. ;)

File: go-controller/pkg/ovn/egressip.go (hunk 1 above)
Comment (kyrtapz) [reply go-controller/pkg/ovn/egressip.go:1590]: since the function is called `addEgressNode` could we just go with `nodeName`?

Comment (tssurya) [reply go-controller/pkg/ovn/egressip.go:1590]: sure yes,

File: go-controller/pkg/ovn/egressip.go (hunk 2 above)
Comment (tssurya) [reply go-controller/pkg/ovn/egressip.go:1642]: done.

**Comment by flavio-fernandes** [comment]
/lgtm
TY for being so awesome, Surya!

File: go-controller/pkg/ovn/egressip.go [hunk 3]
```diff
@@ -2078,19 +2078,15 @@ func (oc *Controller) checkEgressNodesReachability() {
 		}
 		oc.eIPC.allocator.Unlock()
 		for nodeName, shouldDelete := range reAddOrDelete {
-			node, err := oc.watchFactory.GetNode(nodeName)
```
Comment (tssurya) [inline go-controller/pkg/ovn/egressip.go:2081]: Note for reviewers: Removing this logic since its not actually adding any value, then only thing we could see is to try and pull the latest state from kubeapi, but we do have other events for nodes that get processed in case the state of kapi changes, so I don't think this GetNode is very useful here.

File: go-controller/pkg/ovn/obj_retry.go [hunk 4]
```diff
@@ -2,14 +2,15 @@ package ovn
...
 	"fmt"
-	ocpcloudnetworkapi "github.com/openshift/api/cloudnetwork/v1"
 	"math/rand"
 	"net"
 	"reflect"
 	"strings"
 	"sync"
 	"time"
 
+	ocpcloudnetworkapi "github.com/openshift/api/cloudnetwork/v1"
```
Comment (flavio-fernandes) [inline go-controller/pkg/ovn/obj_retry.go:12]: hm... did gofmt do that? interesting.

**Review by flavio-fernandes — COMMENTED** [review]
/lgtm
TY Surya!

File: go-controller/pkg/ovn/obj_retry.go (hunk 4 above)
Comment (tssurya) [reply go-controller/pkg/ovn/obj_retry.go:12]: yup I think so..I didn't :D

**Review by trozet — APPROVED** [review]
